LibNoDaveConnectionLibrary/Communication/LibNoDave/libnodave.net.cs

```diff
@@ -1386,6 +1386,14 @@ public int daveRecieveData(out byte[] data, out byte[] param)
                 else
                     res = daveGetResponse32(pointer);
 
+                //if there was an error getting the response, then fail fast
+                if (res != 0)
+                {

```

Comment:
Yes, because the subsequent calls to libnodave will overwrite the return value from Davegetresponse. This may hide failures.

As in the my case: I erroronously called davegetresponse  which failed with res = -1025 (restimeout). And since getpdu function overwrites the error, and in my case even succeeded since there where in fact data , the function return 0 at the end. 
I just noticed because the function took around  2 seconds per call, because it was in fact timeoutimg.

I think better should fail to uncover mistakes like this.